ros2param/ros2param/verb/dump.py

```diff
@@ -52,8 +52,8 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             default='.',
             help='The absolute path were to save the generated file')
         parser.add_argument(
-            '--print', action='store_true',
-            help='Print generated file in terminal rather than saving a file.')
+            '--write', action='store_true',
+            help='Save parameters to a file, otherwise print to stdout in terminal.')
```

Comment:
I'm wondering if we need the `--write` argument at all.  After all, the user can always redirect the output from stdout to a file (`ros2 param dump /foo > foo.yaml`), and that seems a lot more natural to me.

What do others think?

Reply:
I agree. The redirect does seem much more natural.

Reply:
that requires user to specify the directory w/o `--output-dir`, i think it is still fine by me. but what would you think? can we deprecate `--output-dir` as well?

Reply:
Thanks for the heads up :+1: .
I personally don't mind the change but it is indeed somewhat breaking. If the initial issue is overwritten files, why not simply fail if the file exists (together with an `--force` flag)? It is still changing the current behavior, but less than this proposal.

Reply:
> I personally don't mind the change but it is indeed somewhat breaking. If the initial issue is overwritten files, why not simply fail if the file exists (together with an `--force` flag)? It is still changing the current behavior, but less than this proposal.

We could also go that route.  My thinking was that in the long run, just always printing to stdout and letting the user use the shell redirection would be better.  So we can move that direction for H-Turtle, and then completely remove the old flags for I-Turtle.  After all, part of the reason for having distributions is so we can make "larger" changes like this.

Reply:
I agree with the reasoning. I'm simply wondering how many people will curse us for this change, hopefully not many :D

Reply:
@clalancette @artivis thanks for the comments.

i am good to go with https://github.com/ros2/ros2cli/pull/638#discussion_r636039660, fix coming up soon.

> Put a release note at https://docs.ros.org/en/rolling on the H-Turtle page about this change in behavior (we'll create that H-Turtle release note page early next week).

i will keep that in mind.

Reply:
> I agree with the reasoning. I'm simply wondering how many people will curse us for this change, hopefully not many :D

Yeah, I do appreciate you bringing up the point.  I think this is a very "leafy" utility, in that I don't expect it to be embedded in too many scripts.  I also think that mildly retraining people on how to use it isn't too onerous.  Finally, we aren't really changing the semantic of what is going on, just where it prints.  So I'm willing to take the risk here and get to where we want to be for H-Turtle.